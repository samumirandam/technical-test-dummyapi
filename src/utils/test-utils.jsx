import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import reduxThunk from "redux-thunk";
import { configureStore as configureReduxStore } from "@reduxjs/toolkit";
import { BrowserRouter } from "react-router-dom";

import reducer from "@reducers";

const middlewares = [reduxThunk];
const mockStore = configureStore(middlewares);

const AllTheProviders = ({ children }) => {
  const store = configureReduxStore({
    reducer: reducer,
  });
  const setupStore = { ...store.getState(), ...children.props };
  const compomentStore = mockStore(setupStore);
  return (
    <Provider store={compomentStore}>
      <BrowserRouter>{children}</BrowserRouter>
    </Provider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
