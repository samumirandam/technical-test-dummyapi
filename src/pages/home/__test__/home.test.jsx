import React from "react";
import { render, screen } from "@utils/test-utils";

import Home from "../index";

const defaultProps = {};

const setup = (properties = {}) => {
  const setupStore = { ...defaultProps, ...properties };
  return render(<Home {...setupStore} />, {});
};

describe("Test for Home page component", () => {
  test("Should render without errors", () => {
    setup();
    expect(screen.getByTestId("Home")).toBeTruthy();
  });

  test("Should renders the same component", () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  test("Should render with data", () => {
    const props = {
      postList: {
        data: {
          data: [
            {
              id: "60d21b4667d0d8992e610c85",
              image:
                "https://img.dummyapi.io/photo-1564694202779-bc908c327862.jpg",
              likes: 43,
              tags: ["animal", "dog", "golden retriever"],
              text: "adult Labrador retriever",
              publishDate: "2020-05-24T14:53:17.598Z",
              owner: {
                id: "60d0fe4f5311236168a109ca",
                title: "ms",
                firstName: "Sara",
                lastName: "Andersen",
                picture: "https://randomuser.me/api/portraits/women/58.jpg",
              },
            },
          ],
        },
      },
    };
    setup(props);
    expect(screen.getByText("adult Labrador retriever")).toBeTruthy();
  });
});
