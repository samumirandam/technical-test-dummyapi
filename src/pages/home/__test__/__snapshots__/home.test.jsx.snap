// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Home page component Should renders the same component 1`] = `
<div>
  <section
    class="Home"
    data-testid="Home"
  >
    <section
      class="PostList"
      data-testid="PostList"
    />
  </section>
</div>
`;
